osutil/disks/mapper.go

```diff
@@ -0,0 +1,59 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package disks
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+// CreateLinearMapperDevice creates a linear device mapping of the given device
+// with the given offset and size.
+//
+// The mapper device node is returned.
+func CreateLinearMapperDevice(device, name, uuid string, offset, size uint64) (string, error) {
+	errPrefix := fmt.Sprintf("cannot create mapper %q on %v: ", name, device)
+
+	if offset%512 != 0 {
```

Comment:
where is the 512 requirement here coming from? is that a sector size related thing, if so then this may need to live in gadget if we are to support 4k sector size eventually because you could have a device which aligns to 512 which coesn't align to 4k. It's unclear to me where the 512 requirement comes from so I'm not sure how big of a deal this is or not, maybe it needs to be aligned to 512 for some other reason